common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/OneForOneBlockFetcher.java

```diff
@@ -100,7 +114,14 @@ public void onSuccess(ByteBuffer response) {
           // Immediately request all chunks -- we expect that the total size of the request is
           // reasonable due to higher level chunking in [[ShuffleBlockFetcherIterator]].
           for (int i = 0; i < streamHandle.numChunks; i++) {
-            client.fetchChunk(streamHandle.streamId, i, chunkCallback);
+            if (fetchToDisk) {
+              final File targetFile = new File(".",
```

Comment:
instead of passing a `boolean fetchToDisk`, shall we ask the caller to pass in `Option<File> file`? My concern is that, Spark has a rule about where to write temp files, we can't just write it to the current directory.